regression/testing_tool.py

```diff
@@ -56,43 +88,101 @@ def _initialize_test_case(self):
             for line in fp:
                 self.test_regex.append(line.strip())
 
+class XMLTestCase(BaseTest):
+    """This specialization will parse XML test descriptions"""
+
+    UNSUPPORTED_OPTIONS = ["--timeout", "--memlimit"]
     def __init__(self, test_dir: str, name: str):
-        assert os.path.exists(test_dir)
-        assert os.path.exists(test_dir + "/test.desc")
-        self.name = name
-        self.test_dir = test_dir
-        self._initialize_test_case()
+        super().__init__(test_dir, name)
+
+    def _initialize_test_case(self):
+        root = ET.parse(os.path.join(self.test_dir, "test.desc")).getroot()        
+        self.version: str = root[0].text.strip()
+        self.module: str = root[1].text.strip()
+        self.description: str = root[2].text.strip()
+        self.test_file: str = root[3].text.strip()
+        self.test_args: str = root[4].text.strip()
+        self.test_args: str = root[4].text.strip()
+        # TODO: Multiline regex
+        self.test_regex = [root[5].text.strip()]
+        self.priority = root[6].text.strip()
+        self.execution_type = root[7].text.strip()
+        self.author = root[8].text.strip()
+
+        try:
+            self.test_mode = root[9].text.strip()
+        except:
+            self.test_mode = "CORE"
+        finally:
+            assert self.test_mode in SUPPORTED_TEST_MODES, str(self.test_mode) + " is not supported"
+        assert os.path.exists(os.path.join(self.test_dir, self.test_file))
+        
+    def generate_run_argument_list(self, executable: str):        
+        result = super().generate_run_argument_list(executable)
+        # Some sins were committed into test.desc hack them here
+        try:
+            index = result.index("~/libraries/")
```

Comment:
this could be a relative path from the perspective of this folder, right?

Reply:
This is coming from the test description. I am replacing it from the testing_tool for now